# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: netwerk/protocol/http/HttpChannelParent.cpp
# Commit: 55b215923179
# Full Hash: 55b215923179e7077b48fd7ee3207f89d778059e
# Author: Honza Bambas <honzab.moz@firemni.cz>
# Date: 2019-10-16 16:42:16
# Description:
#   Bug 1586195 - Add a non-null check for mChannel in HttpChannelParent::RecvRedirect2Verify as it can already be legally dropped, r=michal
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D48841
# ==============================================================================

diff -r 4976fa1575dd -r 55b215923179 netwerk/protocol/http/HttpChannelParent.cpp
--- a/netwerk/protocol/http/HttpChannelParent.cpp	Wed Oct 16 07:25:41 2019 +0000
+++ b/netwerk/protocol/http/HttpChannelParent.cpp	Wed Oct 16 07:51:55 2019 +0000
@@ -938,7 +938,9 @@
 
   // If the redirect is vetoed, reason is set on the source (current) channel's
   // load info, so we must carry iver the change.
-  if (aSourceRequestBlockingReason) {
+  // The channel may have already been cleaned up, so there is nothing we can
+  // do.
+  if (MOZ_UNLIKELY(aSourceRequestBlockingReason) && mChannel) {
     nsCOMPtr<nsILoadInfo> sourceLoadInfo = mChannel->LoadInfo();
     if (sourceLoadInfo) {
       sourceLoadInfo->SetRequestBlockingReason(aSourceRequestBlockingReason);
